subsys/bluetooth/host/ecc.c

```diff
@@ -64,8 +64,18 @@ int bt_pub_key_gen(struct bt_pub_key_cb *new_cb)
 		}
 	}
 
-	new_cb->_next = pub_key_cb;
-	pub_key_cb = new_cb;
+	if (new_cb) {
```

Comment:
this suggest that passing NULL new_cb is valid, but since list is cleared after key is generated it makes no sense to call with NULL.  I'd suggest to return error if callback is NULL and remove note about updating key in such case from bt_pub_key_gen() documentation.